pywbem_mock/_baseprovider.py

```diff
@@ -88,6 +76,16 @@ def __init__(self, cimrepository=None, provider_registry=None):
             repository is fully initialized.
 
           provider_registry (dict): Provider registry.
+
+          provider_type (:term:`string`):
+            Keyword defining the type of request the provider will service.
+            The allowed types are instance (keyword 'instance') which responds
+            to the request responder methods defined in InstanceWriteProvider
+            (ex. CreateInstance) and method (keyword 'method') which responds
+            to the InvokeMethod.
+
+          provider_class (:term:`string`):
+            The class for which the provider is registered.
         """
         self.cimrepository = cimrepository
         self.provider_registry = provider_registry
```

Comment:
The class has a property `disable_pull_operations`, and I think the init method should initialize the corresponding instance attribute `self._disable_pull_operations`.

Reply:
Good catch.  Also moved to mainprovider